__tests__/helper.spec.js

```diff
@@ -1,18 +1,167 @@
+import TestHelper from './TestHelper';
 import Helper from '../src/helper';
+import { MosaicService, NamespaceService } from '../src/infrastructure';
+import http from '../src/infrastructure/http';
+import { stub, restore } from 'sinon';
+import { Mosaic, MosaicId, UInt64, NamespaceId, NamespaceName } from 'symbol-sdk';
 
-describe('Helper', () => {
+const mockTestMosaic = {
+	idHex: '6B2E9EAF2632AEC8',
+	namespaceName: 'namespace_mosaic'
+};
+
+describe.only('Helper', () => {
 	describe('hslToRgb should', () => {
 		it('return rgb', () => {
 			// Arrange:
 			const hue = 0.77;
 			const saturation = 0.9;
 			const lightness = 0.8;
 
-			//Act:
+			// Act:
 			const rgb = Helper.hslToRgb(hue, saturation, lightness);
 
-			//Assert:
+			// Assert:
 			expect(rgb).toEqual({R: 215, G: 158, B: 250});
 		});
 	});
+
+	describe('mosaicsFieldObjectBuilder should', () => {
+		beforeEach(() => {
+			stub(MosaicService, 'getMosaics').resolves(Promise.resolve([
+				TestHelper.mockMosaicInfo(mockTestMosaic.idHex, 'TC46AZWUIZYZ2WVGLVEZYNZHSIFAD3AFDPUJMEA', 10, 0)
+			]));
+
+			stub(NamespaceService, 'getMosaicsNames').resolves(Promise.resolve([{
+				names: [new NamespaceName(new NamespaceId(mockTestMosaic.namespaceName), mockTestMosaic.namespaceName)],
+				mosaicId: mockTestMosaic.idHex
+			}]));
+
+			stub(NamespaceService, 'getLinkedMosaicId').resolves(Promise.resolve(new MosaicId(mockTestMosaic.idHex)));
+		});
+
+		afterEach(restore);
+
+		it('returns basic mosaic object', async () => {
+			// Arrange:
+			const mockMosaics = [
+				new Mosaic(new MosaicId(http.networkCurrency.mosaicId), UInt64.fromUint(1)),
+				new Mosaic(new MosaicId(mockTestMosaic.idHex), UInt64.fromUint(4))
+			];
+
+			// Act:
+			const result = await Helper.mosaicsFieldObjectBuilder(mockMosaics);
+
+			// Assert:
+			expect(result).toStrictEqual([
+				{
+					amount: '0.000001',
+					id: new MosaicId(http.networkCurrency.mosaicId).id,
+					mosaicAliasName: http.networkCurrency.namespaceName,
+					mosaicId: http.networkCurrency.mosaicId,
+					rawAmount: UInt64.fromUint(1)
+				},
+				{
+					id: new MosaicId(mockTestMosaic.idHex).id,
+					amount: '4',
+					rawAmount: UInt64.fromUint(4),
+					mosaicId: mockTestMosaic.idHex,
+					mosaicAliasName: [mockTestMosaic.namespaceName]
```

Comment:
why is this array in this object but not previous one ?

Reply:
so mosaic can have more than one alias name, this is the reason I use it as Array.

the previous one is network currency, which is that is not possible to have more than one alias name.

but it looks strange, I should convert it to the array.